.github/workflows/unit.yml

```diff
@@ -0,0 +1,90 @@
+name: unit
+
+on:
+  push:
+    branches: [master]
+  pull_request: # run on all PRs, not just PRs to a particular branch
+
+jobs:
+  # `unit` includes just unit and proto tests.
+  unit:
+    strategy:
+      matrix:
+        node: ['12', '14', '16']
+    runs-on: ubuntu-latest
+    name: node ${{ matrix.node }}
+    env:
+      LATEST_NODE: '16'
+
+    steps:
+    - name: git clone
+      uses: actions/checkout@v2
+      with:
+        # Depth of at least 2 for codecov coverage diffs. See https://github.com/GoogleChrome/lighthouse/pull/12079
+        fetch-depth: 2
+
+    - name: Use Node.js ${{ matrix.node }}
+      uses: actions/setup-node@v1
+      with:
+        node-version: ${{ matrix.node }}
+
+    - name: Set up protoc
+      uses: arduino/setup-protoc@64c0c85d18e984422218383b81c52f8b077404d3
+      with:
+        version: '3.7.1'
+        repo-token: ${{ secrets.GITHUB_TOKEN }}
+
+    - name: Set up Python
+      uses: actions/setup-python@v1
+      with:
+        python-version: 2.7
+    - name: Install Python dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install protobuf==3.7.1
+
+    - run: yarn install --frozen-lockfile --network-timeout 1000000
+
+    - run: yarn test-proto # Run before unit-core because the roundtrip json is needed for proto tests.
+
+    - run: sudo apt-get install xvfb
+
+    - name: yarn unit
+      if: ${{ matrix.node != env.LATEST_NODE }}
```

Comment:
only running coverage on latest node means better coverage (since it'll have the latest v8 coverage updates) and saves ~3 minutes on the unit run for those not doing coverage